ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-matrix-answer/activity-matrix-answer.component.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+
+import {
+  ActivityMatrixQuestionBlock,
+  Group,
+  Option,
+  Question,
+  SelectMode,
+} from '../../../../models/activity/activityMatrixQuestionBlock';
+import { ActivityMatrixAnswerDto } from '../../../../models/activity/activityMatrixAnswerDto';
+
+interface RenderGroup extends Group {
+  colSpan: number;
+  options: Option[];
+}
+
+@Component({
+  selector: 'ddp-activity-matrix-answer',
+  templateUrl: './activity-matrix-answer.component.html',
+  styleUrls: ['./activity-matrix-answer.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ActivityMatrixAnswer implements OnChanges {
+  @Input() block: ActivityMatrixQuestionBlock;
+  @Input() readonly: boolean;
+  @Output() valueChanged = new EventEmitter<ActivityMatrixAnswerDto[]>();
+  renderGroups: RenderGroup[];
+
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
#minor you can alternatevely use input setter

Reply:
Thank you! It's way better approach.